pybamm/models/submodels/interface/kinetics/total_kinetics.py

```diff
@@ -0,0 +1,247 @@
+                }
+            )
+        for reaction_name in reaction_names:
+            j_n_scale = param.n.j_scale
+            j_p_scale = param.p.j_scale
+
+            j_p_av = variables[
+                f"X-averaged positive electrode {reaction_name}"
+                "interfacial current density"
+            ]
+
+            zero_s = pybamm.FullBroadcast(0, "separator", "current collector")
+            j_p = variables[
+                f"Positive electrode {reaction_name}interfacial current density"
+            ]
+
+            if self.half_cell:
+                j = pybamm.concatenation(zero_s, j_p)
+                j_dim = pybamm.concatenation(zero_s, j_p_scale * j_p)
+            else:
+                j_n_av = variables[
+                    f"X-averaged negative electrode {reaction_name}"
+                    "interfacial current density"
+                ]
+                j_n = variables[
+                    f"Negative electrode {reaction_name}interfacial current density"
+                ]
+                j = pybamm.concatenation(j_n, zero_s, j_p)
+                j_dim = pybamm.concatenation(j_n_scale * j_n, zero_s, j_p_scale * j_p)
+
+            if reaction_name not in ["SEI ", "lithium plating "]:
+                j0_p = variables[
+                    f"Positive electrode {reaction_name}exchange current density"
+                ]
+
+                if self.half_cell:
+                    j0 = pybamm.concatenation(zero_s, j0_p)
+                    j0_dim = pybamm.concatenation(zero_s, j_p_scale * j0_p)
+                else:
+                    j0_n = variables[
+                        f"Negative electrode {reaction_name}exchange current density"
+                    ]
+                    j0 = pybamm.concatenation(j0_n, zero_s, j0_p)
+                    j0_dim = pybamm.concatenation(
+                        j_n_scale * j0_n, zero_s, j_p_scale * j0_p
+                    )
+                new_variables.update(
+                    {
+                        f"{reaction_name}interfacial ".capitalize()
+                        + "current density": j,
+                        f"{reaction_name}interfacial ".capitalize()
+                        + "current density [A.m-2]": j_dim,
+                        f"{reaction_name}interfacial ".capitalize()
+                        + "current density per volume [A.m-3]": i_typ / L_x * j,
+                        f"{reaction_name}exchange ".capitalize()
+                        + "current density": j0,
+                        f"{reaction_name}exchange ".capitalize()
+                        + "current density [A.m-2]": j0_dim,
+                        f"{reaction_name}exchange ".capitalize()
+                        + "current density per volume [A.m-3]": i_typ / L_x * j0,
+                    }
+                )
+
+            # Sum variables
+            if pybamm.xyz_average(j_p).id == pybamm.Scalar(0).id:
+                a_p = j_p  # zero
```

Comment:
can you add a more detailed comment here? on first read setting `a` to zero seems wrong